packages/components/psammead-media-player/src/Canonical/useTimeout.jsx

```diff
@@ -0,0 +1,33 @@
+import { useRef, useEffect } from 'react';
+
+/**
+ * useTimeout Hook.
+ * @param {function} callback A callback function that's passed `true` if the timer
+ *                            completes, and `false` if it's interrupted.
+ * @param {object} iframeRef A React Ref. providing access to an iframe DOM node.
+ * @param {number} timeout The number of milliseconds until the timer completes.
+ */
+const useTimeout = (callback, iframeRef, timeout) => {
+  // Persist a reference to the timer across renders. This is used to clearTimeout.
+  const timer = useRef(null);
+
+  const handleLoad = () => {
+    clearTimeout(timer.current);
+    callback(false);
+  };
+
+  // The function we pass to useEffect will run on mount, and the one we return will
+  // run on unmount. We don't do anything if the value of `timeout` can't be used.
+  // eslint-disable-next-line consistent-return
+  useEffect(() => {
+    if (timeout > 0) {
+      iframeRef.current.addEventListener('load', handleLoad, { once: true });
```

Comment:
The value of `.current` is guaranteed to be defined at this point, right?

Reply:
If `iframeRef` is created as documented (as a React Ref), it will be, but perhaps this should be more defensively written, because that can't be guaranteed. I deliberated over this for a while and appreciate you raising it. It should probably throw if `current` is falsy. I'll make this change now.